perfkitbenchmarker/linux_benchmarks/dpb_wordcount_benchmark.py

```diff
@@ -0,0 +1,173 @@
+from perfkitbenchmarker.providers.gcp import gcp_dpb_dataproc
+from perfkitbenchmarker.providers.gcp import gcp_dpb_dataflow
+
+BENCHMARK_NAME = 'dpb_wordcount_benchmark'
+
+BENCHMARK_CONFIG = """
+dpb_wordcount_benchmark:
+  description: Run word count on dataflow and dataproc
+  dpb_service:
+    service_type: dataproc
+    worker_group:
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-1
+          boot_disk_size: 500
+        AWS:
+          machine_type: m3.medium
+      disk_spec:
+        GCP:
+          disk_type: nodisk
+        AWS:
+          disk_size: 500
+          disk_type: gp2
+    worker_count: 2
+"""
+
+word_count_configuration = dict(
+    [
+        ('dataproc', (gcp_dpb_dataproc.SPARK_SAMPLE_LOCATION,
+                      'org.apache.spark.examples.JavaWordCount',
+                      BaseDpbService.SPARK_JOB_TYPE)),
+        ('dataflow', (gcp_dpb_dataflow.DATAFLOW_WC_JAR,
+                      'com.google.cloud.dataflow.examples.WordCount',
+                      BaseDpbService.DATAFLOW_JOB_TYPE)),
+        ('emr', (aws_dpb_emr.SPARK_SAMPLE_LOCATION,
+                 'org.apache.spark.examples.JavaWordCount',
+                 BaseDpbService.SPARK_JOB_TYPE))
+    ]
+)
+
+flags.DEFINE_string('dpb_wordcount_input', None, 'Input for word count')
+flags.DEFINE_enum('dpb_wordcount_fs', BaseDpbService.GCS_FS,
+                  [BaseDpbService.GCS_FS, BaseDpbService.S3_FS],
+                  'File System to use for the job output')
+flags.DEFINE_string('dpb_wordcount_out_base', None,
+                    'Base directory for word count output')
+
+FLAGS = flags.FLAGS
+
+
+def GetConfig(user_config):
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def CheckPrerequisites():
+  """Verifies that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.data.ResourceNotFound: On missing resource.
+  """
+  if (FLAGS.dpb_wordcount_input is None and
+          FLAGS.dpb_wordcount_fs != BaseDpbService.GCS_FS):
+    raise ResourceNotFound
+
+
+def Prepare(benchmark_spec):
+  pass
+
+
+def Run(benchmark_spec):
+
+  # Configuring input location for the word count job
+  if FLAGS.dpb_wordcount_input is None:
+    input_location = gcp_dpb_dataflow.DATAFLOW_WC_INPUT
+  else:
+    input_location = '{}://{}'.format(FLAGS.dpb_wordcount_fs,
+                                      FLAGS.dpb_wordcount_input)
+
+  # Get handle to the dpb service
+  dpb_service = benchmark_spec.dpb_service
+
+  # Create a file handle to contain the response from running the job on
+  # the dpb service
+  stdout_file = tempfile.NamedTemporaryFile(suffix='.stdout',
+                                            prefix='dpb_wordcount_benchmark',
+                                            delete=False)
+  stdout_file.close()
+
+  # Switch the parameters for submit job function of specific dpb service
+  job_arguments = []
+  jarfile, classname, job_type = _GetJobArguments(dpb_service.SERVICE_TYPE)
```

Comment:
Moved the validation to Decoder since the checks to be performed are generic and not specific to the word count benchmark.
Essentially, every dataflow job will require an executable jar and a staging location so making them part of the dataflow implementation